### PR TITLE
Support nv25 and builtin-actors v16

### PR DESCRIPTION
**Pull request by ziscky**
# Support for Lotus nv25 and builtin-actors v16 

## Overview
This PR introduces support for builtin-actors v16 for NV25 and fixes a significant bug with calibration network version handling. 

### New Features
- Added support for builtin-actors v16
- Implemented version iterator to handle different latest versions for mainnet and calibration networks
- Updated missing versions across various actors
- Enhanced version mapping support

### Bug Fixes
- Fixed mainnet edge case issues
- Resolved calibration network mapping issues
- Updated various actor implementations for better compatibility

### Testing Improvements
- Increased test coverage across all actor tests
- Updated test suite to use new LatestVersion method
- Enhanced test utilities and improved test cases

### Code Maintenance
- Updated dependencies (go.mod and go.sum)
- Performed code cleanup and organization
- Added new version iterator utility (tools/version_iterator.go)

## Modified Components
The changes affect multiple components including:
- Account, Cron, Datacap, EAM, EVM actors
- Init, Market, Miner actors
- Multisig, Payment Channel actors
- Power, Reward, Verified Registry actors
- Testing utilities and version mapping tools

## Testing
All tests have been updated and verified to pass with the new changes. Test coverage has been improved across the codebase.

### CODE REVIEW

File: actors/v2/market/market.go [hunk 1]
```diff
@@ -58,37 +60,49 @@ func (*Market) StartNetworkHeight() int64 {
 	return tools.V1.Height()
 }
 
-func (*Market) Methods(_ context.Context, network string, height int64) (map[abi.MethodNum]nonLegacyBuiltin.MethodMeta, error) {
+func (m *Market) Methods(_ context.Context, network string, height int64) (map[abi.MethodNum]nonLegacyBuiltin.MethodMeta, error) {
+	if height == 2523454 {
+		fmt.Printf("network: %s height: %d\n", network, height)
+	}
```
Comment (emmanuelm41) [inline actors/v2/market/market.go:66]: debug line here

**Review by emmanuelm41 — APPROVED** [review]
It is simply amazing to have this shipped and ready to be activated the moment the network height is reached. I am in love with this. Just leave a question: Why tests have changed SO much?  I see a really big change on tests now.

File: tools/version_iterator.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+func NewVersionIterator(from version, network string) *VersionIterator {
+	v := &VersionIterator{
+		network: network,
+		from:    from,
+	}
+	v.Reset()
+	return v
+}
+
+// Next moves to and returns the next version
```
Comment (JulianToledano) [inline tools/version_iterator.go:22]: isn't this returning the current and advancing next?

File: tools/version_mapping.go [hunk 3]
```diff
@@ -56,65 +76,87 @@ func ParseRawNetworkName(network string) string {
...
+
+func isSupported(network string, height int64, iter *VersionIterator) bool {
+	v, ok := iter.Peek()
+	if !ok {
+		return false
+	}
 	if network == CalibrationNetwork {
-		return checkCalibrationEdgeCases(network, height, v)
+		return checkCalibrationEdgeCases(network, height, iter)
 	} else {
```
Comment (JulianToledano) [inline tools/version_mapping.go:90]: we can get rid of this else

**Review by JulianToledano — COMMENTED** [review]
LGTM 🚀

**Review by JulianToledano — APPROVED** [review]

File: actors/v2/market/market.go (hunk 1 above)
Comment (ziscky) [reply actors/v2/market/market.go:66]: Yeah the test to verify all actor methods here: https://github.com/Zondax/fil-parser/blob/f76092e03478f9c5e47ef0f09a798a942e0e7de7/actors/v2/actors_test.go#L373

are now covered by this, since all actors have a `Methods()` fn:
https://github.com/Zondax/fil-parser/blob/58241d50b985ead554f97d86953b2438ac838686/actors/v2/actors_test.go#L65
